### ISSUE TITLE
r.json ignores trailing characters

### ISSUE THREAD

**Issue by AtnNn** [rethinkdb/rethinkdb#3810]
```
>>> r.json('1+2')
1
```

**Comment by danielmewes** [comment]
I have a branch where I've started replacing cJSON with RapidJSON. Might be easier to fix there.

**Comment by AtnNn** [comment]
When do you plan on merging that branch? I have a simple fix for cJSON, it just doesn't check the return value.

**Comment by danielmewes** [comment]
@AtnNn Oh ok. Then go ahead. I will not merge my branch until after 2.0.

**Comment by danielmewes** [comment]
(ideally write a regression test so I don't mess it up again)

**Comment by AtnNn** [comment]
Apparently all ASCII control characters are whitespace.

```
/* Utility to jump whitespace and cr/lf */
static const char *skip(const char *in) {while (in && *in && (unsigned char)*in<=32) in++; return in;}
```

```
>>> r.json('\x01"hello"')
"hello"
```

**Comment by AtnNn** [comment]
Fixed in 05f04d1
